docs/basics/index-lifecycle.md

```diff
@@ -0,0 +1,39 @@
+
+Index Lifecycle
+======================
+
+JanusGraph uses only indexes which have status `ENABLED`. 
+When the index is created it will not be used by JanusGraph until it is enabled. 
+After the index is build you should wait until it is registered (i.e. available) by JanusGraph:
+```java
+//Wait for the index to become available (i.e. wait for status REGISTERED)
+ManagementSystem.awaitGraphIndexStatus(graph, "nameAndAgeIndex").call();
+```
+
+After the index is registered we should either enable the index 
+(if we are sure that the current data should not be indexed by the newly created index) or we should reindex current data 
+so that it would be available in the newly created index.
+```java
+//Reindex the existing data
+mgmt = graph.openManagement();
+mgmt.updateIndex(mgmt.getGraphIndex("nameAndAgeIndex"), SchemaAction.REINDEX).get();
+mgmt.commit();
+```
+
+Index states and transitions
+-----------
+
+![States and transitions](index-lifecycle.png)
+
+## States (SchemaStatus)
```

Comment:
I think it would be good to have a short sentence as an introduction for this enumeration, maybe something like: _An index can be in one of the following states:_